### PR TITLE
Jest [rebase, fix errors, tweak config, misc updates]

### PR DESCRIPTION
**Pull request by justingreenberg**
@dmitriiabramov this will bring your branch up to date with master, fixes lint errors, etc. 

all automated tests should pass 🙏

### CODE REVIEW

**Comment by boujeepossum** [comment]
than you so much for doing this!

that's a lot of commits. let me try to pull it and see what's going on :)

**Comment by justingreenberg** [comment]
no problem! yea this was a little bit out of sync with development branch lol

File: package.json [hunk 1]
```diff
@@ -35,15 +35,15 @@
...
     "lint:js": "npm run lint:eslint -- . ",
     "lint:css": "stylelint ./app/**/*.css",
     "lint:staged": "lint-staged",
     "pretest": "npm run test:clean && npm run lint",
     "test:clean": "rimraf ./coverage",
-    "test": "cross-env NODE_ENV=test jest",
-    "test:watch": "cross-env NODE_ENV=test jest",
-    "coveralls": "cat ./coverage/lcov/lcov.info | coveralls"
+    "test": "cross-env NODE_ENV=test jest --no-cache --coverage",
+    "test:watch": "cross-env NODE_ENV=test jest --no-cache --coverage",
```
Comment (andykenward) [inline package.json]: change to
`"test:watch": "cross-env NODE_ENV=test jest --watch --coverage",`

File: package.json [hunk 2]
```diff
@@ -200,67 +210,69 @@
...
     "eventsource-polyfill": "0.9.6",
     "exports-loader": "0.6.3",
-    "extract-text-webpack-plugin": "1.0.1",
+    "extract-text-webpack-plugin": "2.0.0-beta.4",
     "file-loader": "0.9.0",
     "html-loader": "0.4.3",
     "html-webpack-plugin": "2.22.0",
     "image-webpack-loader": "2.0.0",
     "imports-loader": "0.6.5",
     "jest": "^15.1.1",
```
Comment (andykenward) [inline package.json]: remove `jest` as you have `jest-cli` installed to now

Looking at the repos for both. `jest` is just a wrapper for `jest-cli`. So go with `jest-cli`

[jest-cli](https://github.com/facebook/jest/blob/ec782b567bf343d9d278cd79265a1af2bfc3e83c/packages/jest-cli/package.json)

[jest](https://github.com/facebook/jest/blob/ec782b567bf343d9d278cd79265a1af2bfc3e83c/packages/jest/package.json)

**Review by andykenward — CHANGES_REQUESTED** [review]
Just some little changes I suggest. 

Also no longer need `internals/webpack/webpack.test.babel.js`

(still getting use to this new review process, not sure about it)

**Comment by justingreenberg** [comment]
@andykenward thank you! i just pushed up the changes you suggested in 8221fb3... everything seems to be working as expected 👍 

edit: was having some coverage issues after removing `no-cache` flag, put it back and changed watch to `watchAll` and we're back in business

File: package.json (hunk 1 above)
Comment (boujeepossum) [reply package.json]: i'm not sure how useful `--coverage` is when used with `--watch` right now. It'll report only partial coverage + `0`s for all untested files.

Comment (justingreenberg) [reply package.json]: that makes sense now, i had to pass `--watchAll` to get full coverage report...

https://github.com/dmitriiabramov/react-boilerplate/pull/1/commits/88a41e26bcfa68c79bdc06fc107be22efc5fd08e#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R45

**Pull request closed without merging**